### PR TITLE
DRAFT: Workaround for mellanox errors in kernel lockdown mode

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -693,3 +701,45 @@ func isSwitchdev(name string) bool {
...
+	return false
+}
+
+// IsKernelLockdownMode returns true when kernel lockdown mode is enabled
+func IsKernelLockdownMode() bool {
+	// Check 2 path to allow using with chroot
+	for _, path := range []string{"/host/sys/kernel/security/lockdown", "/sys/kernel/security/lockdown"} {
+		out, err := RunCommand("cat", path)
+		glog.V(2).Infof("IsKernelLockdownMode(): %s, %+v", out, err)
+		if err == nil && strings.Contains(out, "[integrity]") {
```
Comment (yuvalk) [inline pkg/utils/utils.go:725]: actually both `[integrity]` and `[confidentiality]` are considered lockdown
so better search for anything other than `[none]`

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 2]
```diff
@@ -70,6 +69,10 @@ func (p *MellanoxPlugin) OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeSt
 // OnNodeStateChange Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
 func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
 	glog.Info("mellanox-Plugin OnNodeStateChange()")
+	if utils.IsKernelLockdownMode() {
```
Comment (SchSeba) [inline pkg/plugins/mellanox/mellanox_plugin.go:72]: we should return and error here so we don't continue to the apply

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 3]
```diff
@@ -173,6 +176,10 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 
 // Apply config change
 func (p *MellanoxPlugin) Apply() error {
+	if utils.IsKernelLockdownMode() {
```
Comment (SchSeba) [inline pkg/plugins/mellanox/mellanox_plugin.go:179]: we don't need this lets stop in the `OnNodeStateChange`

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -136,7 +138,13 @@ func DiscoverSriovDevices() ([]sriovnetworkv1.InterfaceExt, error) {
 // SyncNodeState Attempt to update the node state to match the desired state
 //
 func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
+	if IsKernelLockdownMode() && hasMellanoxInterfacesInSpec(newState) {
```
Comment (SchSeba) [inline pkg/utils/utils.go:141]: I don't think we need it here

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -693,3 +701,45 @@ func isSwitchdev(name string) bool {
...
+			}
+		}
+	}
+	return false
+}
+
+// IsKernelLockdownMode returns true when kernel lockdown mode is enabled
+func IsKernelLockdownMode() bool {
+	// Check 2 path to allow using with chroot
+	for _, path := range []string{"/host/sys/kernel/security/lockdown", "/sys/kernel/security/lockdown"} {
```
Comment (SchSeba) [inline pkg/utils/utils.go:722]: I think we can drop the two links as we only check now outside the chroot

**Review by SchSeba — COMMENTED** [review]
Thanks for the PR I add a few comments

**Comment by SchSeba** [comment]
closing this one in favor of https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/103

/close

**Pull request closed without merging**